### PR TITLE
fix: for gainsight px only new users to have default signUp date

### PR DESCRIPTION
**Pull request by shrouti1507**
## What are the changes introduced in this PR?

resolves INT-1179
We added the default signup date and create date for all the users ( existing / nonexisting). For any existing users, even if the message.traits object was not having signUpDate or createdDate properties it was being populated from message.timestamp / message.original timestamp field and every time the signUp and createdDate was being updated.

In this PR we are fixing the issues. We will only fallback to originalTimestamp / timestamp values if the user is new.

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Type of change

 **bug-fix**

- Existing capabilities/behavior

We added the default signup date and create date for all the users ( existing / nonexisting). For any existing users, even if the message.traits object was not having signUpDate or createdDate properties it was being populated from message.timestamp / message.original timestamp field and every time the signUp and createdDate was being updated.

- New capabilities/behavior
In this PR we are fixing the issues. We will only fallback to originalTimestamp / timestamp values if the user is new.

### Any technical or performance related pointers to consider with the change?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### If the PR has changes in more than 10 files, please mention why the changes were not split into multiple PRs.

N/A

### If multiple linear tasks are associated with the PR changes, please elaborate on the reason:

N/A

<hr>

### Developer checklist

- [x] **No breaking changes are being introduced.**

- [ ] Are all related docs linked with the PR?

- [x] Are all changes manually tested?

- [ ] Does this change require any documentation changes?

- [x] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Review by yashasvibajpai — APPROVED** [review]

File: src/v0/destinations/gainsight_px/transform.js [hunk 1]
```diff
@@ -56,6 +60,16 @@ const identifyResponseBuilder = async (message, { Config }) => {
     payload.firstName = fName;
     payload.lastName = lName;
   }
+  // Only for the case of new user creation, if signUpDate is not provided in traits, timestamp / originalTimestamp is mapped
+  if (!isPresent && !payload.signUpDate) {
```
Comment (koladilip) [inline src/v0/destinations/gainsight_px/transform.js:64]: what is Present is it userID or traits? make it more clearer

**Comment by Gauravudia** [comment]
LGTM

File: src/v0/destinations/gainsight_px/transform.js (hunk 1 above)
Comment (shrouti1507) [reply src/v0/destinations/gainsight_px/transform.js:64]: at line number 54:   
`const { success: isPresent } = await objectExists(userId, Config, 'user');`

using the userID we are checking if the user is already present. and keeping the status in isPresent

Comment (koladilip) [reply src/v0/destinations/gainsight_px/transform.js:64]: can we rename as isUserPresent?

Comment (shrouti1507) [reply src/v0/destinations/gainsight_px/transform.js]: addressed

**Review by koladilip — APPROVED** [review]

**Review by yashasvibajpai — APPROVED** [review]

**Pull request merged**